frontend/components/RequestCrawler.php

```diff
@@ -0,0 +1,60 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: jeniatr
+ * Date: 24.01.17
+ * Time: 23:56
+ */
+
+namespace frontend\components;
+
+use app\models\Tovar;
+use app\models\SaveFile;
+use yii\helpers\Json;
+
+class RequestCrawler
+{
+
+    public $path;
+    public $type;
+
+
+    public function export()
+    {
+        $table = new Tovar();
+        $model = new SaveFile();
+        $fileName = $table::tableName();
+        $data = $table::find()->asArray()->all();
+
+        if ('JSON' == $this->type) {
+            $model->save($this->path, $this->type, $fileName, $this->enJson($data));
+        }
+
+        if ('XML' == $this->type) {
+            $model->save($this->path, $this->type, $fileName, $this->enXml($data));
+        }
+
+    }
+
+    private function enJson($data)
+    {
+//        $f= new FileHelper();
+//        $r=$f::findFiles(__DIR__);
+//        var_dump($r);
+//        exit;
+
+        return Json::encode($data);
```

Comment:
This should be performed by other service called "serializer". Please create two serializers and inject the needed serializer in the configuration.